### PR TITLE
Annotation of Citation Links

### PR DESCRIPTION
**Pull request by nickmwangemi**
This PR contains logic for annotating citation links

### CODE REVIEW

File: peachjam_api/urls.py [hunk 1]
```diff
@@ -9,4 +9,9 @@
 
 urlpatterns = [
     path("", include(router.urls)),
+    path(
+        "documents/<int:pk>/enrichments/citation-links",
+        views.CitationLinkDetail.as_view(),
```
Comment (longhotsummer) [inline peachjam_api/urls.py:14]: The `CitationLinkDetail` view doesn't apply the `pk` filter to anything, so this view is no different to a normal listing view -- which is fine. Perhaps simply add a `citation-links` endpoint to the `router` above, that is a normal CRUD view? I don't see a reason right now to have two separate views.

File: peachjam_api/serializers.py [hunk 2]
```diff
@@ -54,3 +54,17 @@ class Meta:
...
+
+
+class CoreDocumentSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = CoreDocument
+        fields = "__all__"
+
+
+class CitationLinkSerializer(serializers.ModelSerializer):
+    document = CoreDocumentSerializer(read_only=True)
```
Comment (longhotsummer) [inline peachjam_api/serializers.py:66]: let's make this just the document id

File: peachjam_api/urls.py (hunk 1 above)
Comment (nickmwangemi) [reply peachjam_api/urls.py:14]: Done.

File: peachjam_api/serializers.py (hunk 2 above)
Comment (nickmwangemi) [reply peachjam_api/serializers.py:66]: Done.

**Review by longhotsummer — APPROVED** [review]

**Pull request merged**